internal/coreinternal/processor/filtermetric/filtermetric.go

```diff
@@ -15,53 +15,168 @@
 package filtermetric // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filtermetric"
 
 import (
+	"fmt"
+
 	"go.opentelemetry.io/collector/model/pdata"
+	conventions "go.opentelemetry.io/collector/model/semconv/v1.6.1"
 	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterconfig"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filtermatcher"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterset"
+)
+
+type FilterType int
+
+const (
+	INCLUDE FilterType = iota + 1
+	EXCLUDE
 )
 
 type Matcher interface {
 	MatchMetric(metric pdata.Metric) (bool, error)
 }
+type AttrMatcher interface {
+	MatchWholeMetric(metric pdata.Metric, resource pdata.Resource, library pdata.InstrumentationLibrary, filterType FilterType) (bool, error)
+	MatchAttributes(atts pdata.AttributeMap, resource pdata.Resource, library pdata.InstrumentationLibrary) bool
+	ChecksAttributes() bool
+}
+
+// propertiesMatcher allows matching a metric against various metric properties.
+type propertiesMatcher struct {
+	filtermatcher.PropertiesMatcher
+
+	// Service names to compare to.
+	serviceFilters filterset.FilterSet
+
+	// Span names to compare to.
+	nameFilters filterset.FilterSet
+}
 
-// NewMatcher constructs a metric Matcher. If an 'expr' match type is specified,
-// returns an expr matcher, otherwise a name matcher.
 func NewMatcher(config *MatchProperties) (Matcher, error) {
 	if config == nil {
 		return nil, nil
 	}
-
 	if config.MatchType == Expr {
 		return newExprMatcher(config.Expressions)
 	}
 	return newNameMatcher(config)
 }
 
+// NewMatcher constructs a metric Matcher. If an 'expr' match type is specified,
```

Comment:
The [convention for Golang](https://go.dev/blog/godoc) is to begin the comment with a name of the element it describes. So `NewAttrMatcher` here and so forth